corehq/apps/cloudcare/static/cloudcare/js/form_entry/fullform-ui.js

```diff
@@ -183,10 +183,61 @@ function Form(json) {
     Container.call(self, json);
     self.submitText = ko.observable('Submit');
 
+    self.currentIndex = ko.observable(0);
+    self.atLastIndex = ko.observable(false);
+
+    var _updateIndexCallback = function (ix, isAtLastIndex) {
+        self.currentIndex(ix);
+        self.atLastIndex(isAtLastIndex);
+    };
+
+    self.showInFormNavigation = ko.observable(
+        self.displayOptions.oneQuestionPerScreen !== undefined
+        && self.displayOptions.oneQuestionPerScreen() === true
+    );
+
+    self.isCurrentRequiredSatisfied = ko.computed(function () {
+        if (!self.showInFormNavigation()) return true;
+        return (self.children()[0].answer() === null && !self.children()[0].required())
+            || self.children()[0].answer() !== null;
+    });
+
+    self.enableNextButton = ko.computed(function () {
+        if (!self.showInFormNavigation()) return false;
+        return self.showInFormNavigation()
+            && self.isCurrentRequiredSatisfied()
+            && self.children()[0].isValid()
```

Comment:
should this not be `_.all(self.children(), function(child) { return child.isValid() })`